save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt

```diff
@@ -97,6 +98,44 @@ class DemoManagerController(
                 .toBodilessEntity()
         }
 
+    @PostMapping("/{organizationName}/{projectName}/upload-file")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
+    @Parameters(
+        Parameter(name = "organizationName", `in` = ParameterIn.PATH, description = "name of saveourtool organization", required = true),
+        Parameter(name = "projectName", `in` = ParameterIn.PATH, description = "name of saveourtool project", required = true),
+        Parameter(name = "version", `in` = ParameterIn.QUERY, description = "version to attach the file to", required = true),
+    )
+    @Operation(
+        method = "POST",
+        summary = "Attach file to demo.",
+        description = "Attach file to demo.",
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully added demo.")
+    fun uploadFile(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @RequestParam version: String,
+        @RequestPart file: FilePart,
+        authentication: Authentication,
+    ): Mono<Long> = blockingToMono {
+        projectService.findByNameAndOrganizationNameAndCreatedStatus(
```

Comment:
maybe add permission validation here instead of `@PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")`: `findWithPermissionByNameAndOrganization`

Reply:
This is what Alexey is doing right now

Reply:
Yes, I will add this after the merge of this PR in my PR from Issue #1676